src/procedural-macros.md

```diff
@@ -1,24 +1,546 @@
 ## Procedural Macros
 
 *Procedural macros* allow creating syntax extensions as execution of a function.
-Procedural macros can be used to implement custom [derive] on your own
-types. See [the book][procedural macros] for a tutorial.
+Procedural macros currently come in one of three flavors:
 
-Procedural macros involve a few different parts of the language and its
-standard libraries. First is the `proc_macro` crate, included with Rust,
-that defines an interface for building a procedural macro. The
-`#[proc_macro_derive(Foo)]` attribute is used to mark the deriving
-function. This function must have the type signature:
+* Custom macros - `my_macro!(...)`
+* Custom derive - `#[derive(MyTrait)]`
+* Custom attributes - `#[my_attribute]`
```

Comment:
Let's kill the word "custom". It means nothing. The word should not appear on this page. We don't call things "custom crates" and "custom functions" and "custom structs".

I would write:

- Function-like macros / bang macros
- Derive macros
- Attribute macros

Reply:
SGTM.